src/components/ComicBook/MathQuestions.tsx

```diff
@@ -0,0 +1,142 @@
+import { Page, Question } from "../../types";
+import { useEffect, useState } from "react";
+import { SyntheticEvent } from "react";
+import comicService from "../../services/comicService";
+import axios from "axios";
+const MathQuestions = ({
+  stateKey,
+  setKey,
+  comicName,
+  comic,
+  page,
+}: {
+  stateKey: string | undefined;
+  setKey: React.Dispatch<React.SetStateAction<string | undefined>>;
+  comicName: string;
+  comic: Page[];
+  page: number;
+}) => {
+  const [questions, setQuestions] = useState<Question[]>([]);
+  const [answers, setAnswers] = useState<string[]>([""]);
+  const [buttonClasses, setButtonClasses] = useState("answer-button");
+
+  useEffect(() => {
+    let newQuestions: Question[] = [];
+    if (comic[page].questionList) {
+      const l = comic[page].questionList.length;
+      for (let q = 0; q < l; q++) {
+        const question = comic[page].questionList[q];
```

Comment:
do not use `for` loop with indexing, use `for of`, `.map` or `.forEach`. Use normal `for` when you need to use indexing.